sum.go

```diff
@@ -31,6 +41,23 @@ func Sum(data []byte, code int, length int) (Multihash, error) {
 		d, err = sumSHA3(data)
 	case DBL_SHA2_256:
 		d = sumSHA256(sumSHA256(data))
+	case BLAKE2B:
+		switch length {
+		case 32:
+			out := blake2b.Sum256(data)
+			d = out[:]
+		case 48:
+			out := blake2b.Sum384(data)
+			d = out[:]
+		case 64:
+			out := blake2b.Sum512(data)
+			d = out[:]
+		default:
+			return nil, fmt.Errorf("unsupported length for blake2b")
+		}
```

Comment:
even after the above ^ i think we should do what we do for sha-* family, supporting 128, 256, 384, 512.